### PR TITLE
feat(paymaster-proxy): create api-key-service client

### PR DESCRIPTION
**Pull request by jakim929**
- generate types build output for `api-key-service`
- create client on `paymaster-proxy`

### CODE REVIEW

**Comment by jakim929** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/ethereum-optimism/ecosystem/268?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#268** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/ecosystem/268?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#260** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/ecosystem/260?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#256** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/ecosystem/256?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#255** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/ecosystem/255?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#230** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/ecosystem/230?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @jakim929 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: nx.json [hunk 1]
```diff
@@ -24,6 +24,13 @@
       "cache": true,
       "dependsOn": ["^build"]
     },
+    "build:types": {
+      "cache": true,
+      "dependsOn": ["^build:types"]
+    },
+    "typecheck": {
+      "dependsOn": ["^build:types"]
+    },
```
Comment (jakim929) [inline nx.json:33]: ideally we should have separate pipelines for typecheck, and another for build

File: apps/paymaster-proxy/package.json [hunk 2]
```diff
@@ -34,6 +35,8 @@
   "dependencies": {
     "@alchemy/aa-alchemy": "^3.6.1",
     "@alchemy/aa-core": "^3.2.0",
+    "@trpc/client": "^10.45.2",
+    "@trpc/server": "^10.45.2",
```
Comment (tremarkley) [inline apps/paymaster-proxy/package.json:39]: how come the client needs the `@trpc/server` dependency?

**Review by tremarkley — APPROVED** [review]

File: apps/paymaster-proxy/package.json (hunk 2 above)
Comment (jakim929) [reply apps/paymaster-proxy/package.json]: actually just tested and seems to work without. removed

**Pull request merged**